src/lib/features/feature-toggle/feature-toggle-store.ts

```diff
@@ -362,6 +362,11 @@ export default class FeatureToggleStore implements IFeatureToggleStore {
             .where({ name })
             .update({ archived_at: null })
             .returning(FEATURE_COLUMNS);
+
+        await this.db(FEATURE_ENVIRONMENTS_TABLE)
```

Comment:
I think this should be part of the archive method, not revive.
Also I'm not sure whether it should be done at the store level or service level.
Finally since we have write operations on the write model we need to wrap the whole service method in a transaction. If you decide to do it as part of archive and bulk archive then I'd wrap it in a tx in a controller doing the archive.

Reply:
The problem right now with doing it at archive is that already archived features will not get it - We could have a migration that does disables the environments for already archived features, but I would rather not.

Reply:
Good point on the transaction wrapping

Reply:
so maybe both? in the long term it makes more sense to me to do it as part of archive. We'll probably also clean the dependencies in the archive so having one "cleanup" location would be nice. 
what's the problem with a db migration?